src/mam/mod_mam.erl

```diff
@@ -688,7 +695,8 @@ config_metrics(HostType) ->
 
 -spec hooks(jid:lserver()) -> [ejabberd_hooks:hook()].
 hooks(HostType) ->
-    [{user_send_packet, HostType, ?MODULE, user_send_packet, 60},
+    [{disco_local_features, HostType, ?MODULE, add_local_features, 99},
```

Comment:
maybe `{disco_local_features, HostType, ?MODULE, disco_local_features, 99}`

Reply:
This name would be consistent with the hook, but on the other hand `add_local_features` is more descriptive as the features are added to the list... Anyway, I can make the handler names more consistent in part 2.